tests/validation/tests/v3_api/test_node.py

```diff
@@ -1,54 +1,961 @@
 import pytest
+import time
+from .common import create_kubeconfig
+from .common import delete_node
+from .common import get_user_client
+from .common import get_user_client_and_cluster
+from .common import execute_kubectl_cmd
+from .common import random_name
+from .common import random_test_name
+from .common import validate_cluster_state
+from .common import wait_for_condition
+from rancher import ApiError
+from .test_rke_cluster_provisioning import DO_ACCESSKEY
+from .test_rke_cluster_provisioning import evaluate_clustername
+from .test_rke_cluster_provisioning import get_custom_host_registration_cmd
+from .test_rke_cluster_provisioning import HOST_NAME
+from .test_rke_cluster_provisioning import random_node_name
+from .test_rke_cluster_provisioning import rke_config
+from .test_rke_cluster_provisioning import wait_for_cluster_node_count
+from lib.aws import AmazonWebServices
 
-from .common import *  # NOQA
+cluster_detail = {"cluster": None, "client": None}
+cluster_node_template = {"cluster": None, "node_pools": None,
+                         "node_template": None, "do_cloud_credential": None,
+                         "label_value": None, "test_label": None}
+cluster_custom = {"cluster": None, "test_label": None,
+                  "label_value": None, "aws_node": None}
 
-def test_add_node_label():
-    client, cluster = get_global_admin_client_and_cluster()
-    test_label = "foo"
-    nodes = client.list_node(clusterId=cluster.id)
-    assert len(nodes.data) > 0
-    node_id = nodes.data[0].id
+
+def test_node_label_add():
+    create_kubeconfig(cluster_detail["cluster"])
+    client = cluster_detail["client"]
+    cluster = cluster_detail["cluster"]
+    test_label = random_name()
+    label_value = random_name()
+    nodes = client.list_node(clusterId=cluster.id).data
+    assert len(nodes) > 0
+    node_id = nodes[0].id
     node = client.by_id_node(node_id)
 
-    # Make sure there is no test label and add test label
     node_labels = node.labels.data_dict()
-    assert test_label not in node_labels
+    node_labels[test_label] = label_value
+    client.update(node, labels=node_labels)
+
+    # Label should be added
+    wait_for_condition(client, node, check_label_added(test_label), None, 10)
+    validate_label_set_on_node(client, node, test_label, label_value)
+
+     # delete label
+    del node_labels[test_label]
+    client.update(node, labels=node_labels)
 
-    node_labels[test_label] = "bar"
+
+def test_node_label_edit():
+    create_kubeconfig(cluster_detail["cluster"])
+    client = cluster_detail["client"]
+    cluster = cluster_detail["cluster"]
+    test_label = random_name()
+    label_value = random_name()
+    nodes = client.list_node(clusterId=cluster.id).data
+    assert len(nodes) > 0
+    node_id = nodes[0].id
+    node = client.by_id_node(node_id)
```

Comment:
Can we have a method to return 1 node given a cluster ?

Reply:
Done
Added a function: `client, node = get_node_details()`